### PR TITLE
Feat/autorenew invite

### PR DESCRIPTION
**Pull request by marie-j**
### Description of the change

Add component to invite customer to switch their service renewal into automatic 

### Needs 
- [x] https://github.com/ovh-ux/ng-payment-method release 

### Related
https://github.com/ovh-ux/ovh-manager-web/pull/853

ref: MANAGER-2002

### CODE REVIEW

File: src/autorenew-invite/autorenew-invite.constant.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export const SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW = ['ASIA', 'AU', 'CA', 'FR', 'QC', 'SG', 'WE', 'WS'];
```
Comment (marie-j) [inline src/autorenew-invite/autorenew-invite.constant.js]: Waiting for confirmation this is the correct list

Comment (cbourgois) [reply src/autorenew-invite/autorenew-invite.constant.js]: Should add line break.

File: src/autorenew-invite/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import angular from 'angular';
+import ovhAngularPaymentMethod from '@ovh-ux/ng-payment-method';
+import translate from 'angular-translate';
+import translateAsyncLoader from '@ovh-ux/translate-async-loader';
+
```
Comment (cbourgois) [inline src/autorenew-invite/index.js]: Should import `ovh-angular-user-pref`  and declare it as module dependency.

Same for `ovh-ui-angular`

**Comment by ovh-cds** [comment]
CDS Report build#37.0 ✘  
* build
  * build ✔ 
  * test ✘

File: src/autorenew-invite/autorenew-invite.component.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<oui-message data-type="info"
+             data-ng-if="$ctrl.serviceInfos && $ctrl.isAutorenewAllowed && !$ctrl.isInAutorenew() && !$ctrl.isExpired() && $ctrl.shouldBotherWithAutorenew()"
+             data-on-dismissed="$ctrl.stopBotheringWithAutorenew()">
+    <span data-translate="autorenew_alert" data-translate-values="{ product: ('autorenew_alert_' + $ctrl.productType | translate) }"></span>
+    <a class="oui-link"
+       data-ng-if="$ctrl.hasPaymentMean"
+       href="{{:: $ctrl.autorenewUrl }}"
```
Comment (FredericEspiau) [inline src/autorenew-invite/autorenew-invite.component.html]:
```suggestion
        data-ng-href="{{:: $ctrl.autorenewUrl }}"
```

`href` is a bit of a particular case: https://docs.angularjs.org/api/ng/directive/ngHref

File: src/autorenew-invite/autorenew-invite.component.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+       data-translate="autorenew_activate">
+    </a>
+    <a href="{{:: $ctrl.guideUrl }}"
```
Comment (FredericEspiau) [inline src/autorenew-invite/autorenew-invite.component.html]: `href` is a bit of a particular case: https://docs.angularjs.org/api/ng/directive/ngHref

```suggestion
     <a data-ng-href="{{:: $ctrl.guideUrl }}"
```

File: src/autorenew-invite/autorenew-invite.controller.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    return this.ovhUserPref.getValue(this.preferenceKey)
+      .then((productAutorenewPreferences) => {
+        this.productAutorenewPreferences = productAutorenewPreferences;
+      })
+      .catch(() => {
+        this.productAutorenewPreferences = [];
+      });
+  }
+
+  shouldBotherWithAutorenew() {
```
Comment (FredericEspiau) [inline src/autorenew-invite/autorenew-invite.controller.js]:
```suggestion
   shouldAskAboutAutoRenewSubscription() {
```
?

😄

File: src/autorenew-invite/autorenew-invite.controller.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  isAutorenewAllowed() {
+    return this.OvhApiMe.v6().get().$promise
+      .then(({ ovhSubsidiary }) => {
+        this.isAutorenewAllowed = _.includes(
+          this.WUC_SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW,
+          ovhSubsidiary,
+        );
+      })
+      .catch(() => {
+        this.isAutorenewAllowed = false;
```
Comment (FredericEspiau) [inline src/autorenew-invite/autorenew-invite.controller.js]: Looks like we should display an error message in this case

File: src/autorenew-invite/autorenew-invite.controller.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    this.$q = $q;
+    this.OvhApiMe = OvhApiMe;
+    this.ovhPaymentMethod = ovhPaymentMethod;
+    this.ovhUserPref = ovhUserPref;
+
+    this.WUC_SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW = WUC_SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW;
+  }
+
+  $onInit() {
+    this.preferenceKey = `${this.productType.toUpperCase()}_AUTORENEW_STOP_BOTHER`;
```
Comment (FredericEspiau) [inline src/autorenew-invite/autorenew-invite.controller.js]: Maybe you should validate bindings as `this.productType` could be empty

File: src/autorenew-invite/autorenew-invite.controller.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+  getDisplayConditions() {
+    return this.getProductAutorenewBotherPreferences()
+      .then(() => this.hasPaymentMean());
+  }
+
+  hasPaymentMean() {
+    return this.ovhPaymentMethod.getAllPaymentMethods()
+      .then((paymentMethods) => {
+        this.hasPaymentMean = !_.isEmpty(paymentMethods);
```
Comment (FredericEspiau) [inline src/autorenew-invite/autorenew-invite.controller.js]:
```suggestion
         this.hasPaymentMean = _.isArray(paymentMethods) && !_.isEmpty(paymentMethods);
```

Lodash doesn't like defensive pattern (have you seen any `ul` in Lodash ??) so if `paymentMethods` is a `string`, you would have an unexpected result

File: src/autorenew-invite/autorenew-invite.controller.js (hunk 6 above)
Comment (marie-j) [reply src/autorenew-invite/autorenew-invite.controller.js]: @FredericEspiau I agree that user should know an error has occured,  but I'm not sure it is relevant in that specific case (display a message to inform that we cannot know if we should display another message or not)

**Pull request merged**